src/main/java/org/broadinstitute/consent/http/models/CreateDuosUserRequest.java

```diff
@@ -0,0 +1,27 @@
+package org.broadinstitute.consent.http.models;
+
+import java.util.List;
+import org.broadinstitute.consent.http.enumeration.UserRoles;
+
+public record CreateDuosUserRequest(
+    String displayName, String email, boolean emailPreference, List<UserRole> roles) {
+
+  public void validate() {
```

Comment:
Why not create a canonical constructor that validates?  This way it will be impossible to build a record of type CreateDuosUserRequest without those fields.
```suggestion
public CreateDuosUserRequest(String displayName, String email, boolean emailPreference, List<UserRole> roles) {
    this.displayName = displayName;
    this.email = email;
    this.emailPreference = emailPreference;
    this.roles = roles;
    validate()
}

  public void validate() {
```